spec/std/sprintf_spec.cr

```diff
@@ -407,10 +407,179 @@ describe "::sprintf" do
       end
     end
 
+    context "scientific format" do
+      it "works" do
+        assert_sprintf "%e", 123.45, "1.234500e+2"
+        assert_sprintf "%E", 123.45, "1.234500E+2"
+
+        assert_sprintf "%e", Float64::MAX, "1.797693e+308"
```

Comment:
**question:** Should we test `Float::MIN` as well?

These basic tests have only positive values (except `-0.0`) so maybe could use some more negatives even?